src/scripts/resolver/background.ts

```diff
@@ -0,0 +1,141 @@
+import {Logger} from "../../lib/logger";
+import {getSupportedTlds} from "../../lib/resolver/resolver";
+import {
+  RESOLUTION_REDIRECT_URL,
+  SUPPORTED_DOMAIN_REFRESH_MINUTES,
+} from "../../lib/resolver/types";
+
+export const waitForSupportedDomains = async () => {
+  // start be removing all rules on extension startup
+  await deleteAllRules();
```

Comment:
Now that it's relying on two API calls, we should better check the liveness for those API calls first before deleting all the rules. Or fetch the APIs first, if error then do nothing.